tests/rptest/services/provider_clients/azure_client.py

```diff
@@ -0,0 +1,99 @@
+from rptest.services.provider_clients.client_utils import query_instance_meta
+
+headers = {"Metadata-Flavor": "Microsoft"}
+
+
+class AzureClient:
+    """
+    Azure Client class.
+    Should implement function similar to AWS EC2 client
+    """
+    def __init__(self,
+                 region,
+                 key,
+                 secret,
+                 tenant,
+                 logger,
+                 endpoint=None,
+                 disable_ssl=True):
+
+        self._region = region
+        self._access_key = key
+        self._secret_key = secret
+        self._tenant = tenant
+        self._endpoint = endpoint
+        self._disable_ssl = disable_ssl
+        self._cli = self._make_client()
+
+    def _make_client(self):
+        # TODO: Work on Azure client
+        return None
+
+    @property
+    def project_id(self):
+        return "devprod-cicd-infra"  # HACK
+
+    def create_vpc_peering(self, params, facing_vpcs=True):
+        """
+        Creates two peering connections facing each other
+        """
+        return "hardcode"  # HACK
+
+    def get_vpc_by_network_id(self, network_id, prefix=None, project_id=None):
+        """
+        Get VPC Network
+        """
+        return "hardcode"  # HACK
+
+    def find_vpc_peering_connection(self, state, params):
+        return None
```

Comment:
In general we should ask if it would be better for things to just throw in all of these unimplemented methods. In the past we had a big problem with lots of tests appearing to pass when the underlying method were doing nothing, so the tests were not testing what they claimed. It is clearer and IMO better to just have the tests being failing with a clear exceptin like "bucket disable not implemented for Azure" than just passing out of the method.